src/Components/Facility/FacilityCreate.tsx

```diff
@@ -245,89 +209,68 @@ export const FacilityCreate = (props: FacilityProps) => {
+    routes.getWardByLocalBody,
+    {
+      pathParams: {
+        id: String(localBodyId),
+      },
+      prefetch: localBodyId !== 0,
+    }
   );
 
-  const fetchData = useCallback(
-    async (status: statusType) => {
+  useQuery(routes.getPermittedFacility, {
+    pathParams: {
+      id: facilityId || "",
+    },
+    prefetch: facilityId !== undefined,
+    onResponse: ({ res, data }) => {
       if (facilityId) {
         setIsLoading(true);
-        const res = await dispatchAction(getPermittedFacility(facilityId));
-        if (!status.aborted && res.data) {
+        if (res?.ok && data) {
           const formData = {
-            facility_type: res.data.facility_type,
-            name: res.data.name,
-            state: res.data.state ? res.data.state : 0,
-            district: res.data.district ? res.data.district : 0,
-            local_body: res.data.local_body ? res.data.local_body : 0,
-            features: res.data.features || [],
-            ward: res.data.ward_object ? res.data.ward_object.id : 0,
-            kasp_empanelled: res.data.kasp_empanelled
-              ? String(res.data.kasp_empanelled)
-              : "false",
-            address: res.data.address,
-            pincode: res.data.pincode,
-            phone_number:
-              res.data.phone_number.length == 10
-                ? "+91" + res.data.phone_number
-                : res.data.phone_number,
-            latitude: res.data.latitude || "",
-            longitude: res.data.longitude || "",
-            type_b_cylinders: res.data.type_b_cylinders,
-            type_c_cylinders: res.data.type_c_cylinders,
-            type_d_cylinders: res.data.type_d_cylinders,
-            expected_type_b_cylinders: res.data.expected_type_b_cylinders,
-            expected_type_c_cylinders: res.data.expected_type_c_cylinders,
-            expected_type_d_cylinders: res.data.expected_type_d_cylinders,
-            expected_oxygen_requirement: res.data.expected_oxygen_requirement,
-            oxygen_capacity: res.data.oxygen_capacity,
+            facility_type: data.facility_type ? data.facility_type : "",
+            name: data.name ? data.name : "",
+            state: data.state ? data.state : 0,
+            district: data.district ? data.district : 0,
+            local_body: data.local_body ? data.local_body : 0,
+            features: data.features || [],
+            ward: data.ward_object ? data.ward_object.id : 0,
+            kasp_empanelled: "",
+            address: data.address ? data.address : "",
+            pincode: data.pincode ? data.pincode : "",
+            phone_number: data.phone_number
+              ? data.phone_number.length == 10
+                ? "+91" + data.phone_number
+                : data.phone_number
+              : "",
+            latitude: data.location ? String(data.location.latitude) : "",
+            longitude: data.location
+              ? String(data.location.latitude)
+              : "" || "",
```

Comment:
1. Are you sure this is a nested JSON? Have you tested it?
2. Why is longitude set to latitude?

```suggestion
            latitude: data.latitude,
            longitude: data.longitude,
```